### PR TITLE
Created the first pass of the fhir function documention

### PR DESCRIPTION
**Pull request by JessicaWNava**
This PR adds documentation on FHIR Functions

Test Steps:
N/A

## Changes
- fhir-functions.md filled in
- Other file changes are the result of the linter.

### CODE REVIEW

File: prime-router/docs/getting-started/fhir-functions.md [hunk 1]
```diff
@@ -1,7 +1,69 @@
...
+  parts of the bundle require this to evaluate them. Thankfully, the FHIR path tool will specify when this is 
+  necessary with a message like `Reference to Encounter/1674584645770942000.cffbb057-46bb-4018-8fce-7a4d552ef7e3 - 
+  use resolve() to navigate into it`
+
+
+- To run a function on a resource, run a command like 
+`Bundle.entry.resource.ofType(Patient)[0].contact.telecom.value.GetPhoneNumberCountryCode()`
+
+
+- To call a function on a resource that is set, use `%resource` as shown here: `%resource.GetPhoneNumberCountryCode()`
```
[the remark this answers is not in the document]
Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md:23]: Yes, if it hasn't been changed from the default.

**Comment by JessicaWNava** [comment]
Back to you @thetaurean, thank you for the feedback!

File: prime-router/docs/getting-started/fhir-functions.md [hunk 2]
```diff
@@ -1,7 +1,69 @@
 # FHIR Functions
+FHIR functions are functions that can be run on a bundle via the command line using the fhirpath tool. To use the FHIR
```
[the remark this answers is not in the document]
Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md]: Got it. Changed it to methods, back to you :)

File: prime-router/docs/getting-started/fhir-functions.md [hunk 3]
```diff
@@ -1,7 +1,85 @@
 # FHIR Functions
+FHIR functions are methods that can be run on a bundle (ex. retrieve an age or telephone area code) via the command line using the fhirpath tool. To use the FHIR
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:2]: `FHIR functions are methods that can be run on a bundle (ex. retrieve an age or telephone area code) via the command line using the fhirpath tool.`

It is my understanding that this statement is true, but that's not what FHIR functions are/their main purpose, so it feels weird to have this as the first sentence? I like how you have internal/external functions broken down below. I would add a third section with a heading like `Prime CLI FHIRPath Tool` and move most/all the content here under there. Then you can have a new introduction up here that talks about FHIR functions generally?

Comment (arnejduranovic) [reply prime-router/docs/getting-started/fhir-functions.md:2]: An idea for reorg: I would suggest keeping in this section the following:

> FHIR functions are methods that can be run on a bundle. Currently, there are two places where FHIR functions are created. Both are in a file called CustomFhirPathFunctions and
both extend an interface called FhirPathFunctions. There are two places for these functions because some
are internal functions, meaning that they can only be called from within the code rather than from the command line.

Also, I would maybe avoid listing out every function we have, and would link to the code where they are defined. In the docs here I would focus more on what makes a method a fhir function, how do you define one, how do you invoke one, etc. Although I do think one or two examples could be helpful and are appropriate.

**Review by arnejduranovic — CHANGES_REQUESTED** [review]

File: prime-router/docs/getting-started/fhir-functions.md (hunk 3 above)
Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md:2]: I thought that was what FHIR functions are and their purpose. Will you please elaborate on what you think should be different?

Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md:2]: Made some of the changes but I doubt this addresses everything you were wanting. Please lmk about the definition.

File: prime-router/docs/getting-started/fhir-functions.md [hunk 4]
```diff
@@ -1,7 +1,68 @@
 # FHIR Functions
+FHIR functions are methods that can be run on a bundle in order to extract data from it (ex. retrieve an age or telephone area code)
+Currently, there are two places where FHIR functions are created. Both are in a file called CustomFhirPathFunctions and
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:3]:
```suggestion
. Currently, there are two places where FHIR functions are created. Both are in a file called `CustomFhirPathFunctions` and
```

Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:3]:
```suggestion
Currently, there are two places where FHIR functions are created. Both are in a file called `CustomFhirPathFunctions` and
```

Comment (arnejduranovic) [reply prime-router/docs/getting-started/fhir-functions.md:3]: Probably check other spots too

File: prime-router/docs/getting-started/fhir-functions.md [hunk 5]
```diff
@@ -1,7 +1,68 @@
...
+both extend an interface called FhirPathFunctions. There are two places for these functions because some
+are internal functions, meaning that they can only be called from within the code rather than from the command line.
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:5]: are internal functions, meaning that they can only be called from within the code rather than from the command line **via the [fhir path tool](internal link here).**

File: prime-router/docs/getting-started/fhir-functions.md [hunk 6]
```diff
@@ -1,7 +1,68 @@
...
+
+## How to define a FHIR path function
+1. Determine which file it will go in by reading the [internal](#internal) and [external](#external) sections of this 
+document.
+2. Add the function itself.
+3. Add it to the resolveFunction method. This registers it so the command line knows what to expect and is valid.
+4. Add it to the execute function method. This registers it so the command line knows what to call.
+
+## FHIR Path Tool
+You can run these functions on a bundle using a command line fhirpath tool. To use the FHIR
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:45]:
```suggestion
You can run these functions on a bundle using the prime command line fhirpath tool. To use the FHIR
```

File: prime-router/docs/getting-started/fhir-functions.md (hunk 4 above)
Comment (arnejduranovic) [reply prime-router/docs/getting-started/fhir-functions.md:3]: Also they are NOT both in this file, correct? This file seems to only have `livdTableLookup`? I would probably reword everything after the first sentence, something like "There are two types of FHIR functions: internal and external. Internal functions are blah blah and external functions are blah blah" then you can talk about what files they relate to in their own sections and not up here in the general definition area.

File: prime-router/docs/getting-started/fhir-functions.md [hunk 7]
```diff
@@ -1,7 +1,68 @@
...
+>"value": "DecimalType[31]"  
+>"unit": "year" &nbsp;  
+>"code": "a"  
+>}  
+>Number of results = 1 ----------------------------
+
+## How to define a FHIR path function
+1. Determine which file it will go in by reading the [internal](#internal) and [external](#external) sections of this 
+document.
+2. Add the function itself.
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:40]: Note here what interface or pattern or signature the method must have or implement and why

File: prime-router/docs/getting-started/fhir-functions.md [hunk 8]
```diff
@@ -1,7 +1,68 @@
...
+3. Add it to the resolveFunction method. This registers it so the command line knows what to expect and is valid.
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:41]:
```suggestion
3. Add it to the `resolveFunction` method. This registers it so the command line knows what to expect and is valid.
```

Let's be more consistent with what we put in code blocks (I have a hard time with this too) and try to have every method name in these quotes.

**Review by arnejduranovic — CHANGES_REQUESTED** [review]

File: prime-router/docs/getting-started/fhir-functions.md (hunk 4 above)
Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md:3]: There are two files with that name and they are located in those.

Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md:3]: I can reword it to be more clear.

File: prime-router/docs/getting-started/fhir-functions.md [hunk 9]
```diff
@@ -1,7 +1,70 @@
...
+An example of one of the currently most complicated functions is the convertDateToAge function:
+- convertDateToAge(<optional> timeUnit, <optional> comparisonDate) - The resource must be a date. Since this function 
+required a lot of code to complete, it is broken out into a separate helper function. This is the pattern we hope to 
+use moving forward. This required so much extra code partially because it takes multiple, optional params which can be
+in any order. One can specify whether this value is returned in years, months, or days. 
+If left off, the method assumes years if it is greater than one year, months if it is less than a year but 
+greater than one month, and days if the value is less than one month. There is also an optional param to pass a 
+comparison date if you don't want to get the age based off of how old they are today.
+(timeUnit, comparisonDate) and (comparisonDate, timeUnit) are both acceptable. It returns an age in years, months, or 
+days.Example Output:
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:30]:
```suggestion
days. Example Output: 
```

File: prime-router/docs/getting-started/fhir-functions.md [hunk 10]
```diff
@@ -1,7 +1,70 @@
 # FHIR Functions
+FHIR functions are methods that can be run on a bundle in order to extract data from it (ex. retrieve an age or telephone area code).
```
Comment (arnejduranovic) [inline prime-router/docs/getting-started/fhir-functions.md:2]: We document how to invoke these functions via the FHIR Path tool, but should we also have a section and talk about how else one might invoke these functions? Like from mapping files? Currently the first line of this file is:

`FHIR functions are methods that can be run on a bundle in order to extract data from it
`
That's good to know! But, I think what is missing, crucially, is the why? Why do we want to extract data from a bundle in this way? Presumably the answer lies in how we use them in our mapping files?

**Review by arnejduranovic — CHANGES_REQUESTED** [review]

**Review by arnejduranovic — CHANGES_REQUESTED** [review]

File: prime-router/docs/getting-started/fhir-functions.md [hunk 11]
```diff
@@ -1,5 +1,8 @@
 # FHIR Functions
 FHIR functions are methods that can be run on a bundle in order to extract data from it (ex. retrieve an age or telephone area code). 
+Sometimes this is needed because we need only a part of what is sent over or need data in a different format for us to
```
[the remark this answers is not in the document]
Comment (JessicaWNava) [reply prime-router/docs/getting-started/fhir-functions.md]: Got it, addressed.

**Review by arnejduranovic — APPROVED** [review]
